ddclient.in

```diff
@@ -2849,6 +2867,85 @@ sub nic_dyndns2_update {
     }
 }
 
+######################################################################
+## nic_dinahosting_update
+######################################################################
+sub nic_dinahosting_update {
+     debug("\nnic_dinahosting_update -------------------");
+
+     ## update each configured host
+     foreach my $h (@_) {
+          my $ip = delete $config{$h}{'wantip'};
+          info("setting IP address to %s for %s", $ip, $h);
+          verbose("UPDATE:","updating %s", $h);
+
+          # alexandregz: dinahosting uses subdomain into another param
+          my ($subdomain, $domain) = split(/\./, $h, 2);
+
+          # Set the URL that we're going to update
+          my $url = "https://$config{$h}{'server'}$config{$h}{'script'}";
+          $url  .= "?AUTH_USER=$config{$h}{'login'}";
+          $url  .= "&AUTH_PWD=$config{$h}{'password'}";
+          $url  .= "&hostname=$subdomain";
+          $url  .= "&domain=$domain";
+          $url  .= "&command=Domain_Zone_UpdateTypeA";  # new zones change to Domain_Zone_AddTypeA
+          $url  .= "&ip=";
+          $url  .= $ip            if $ip;
+
+          # Try to get URL
+          my $reply = geturl({ proxy => opt('proxy'), url => $url });
+
+          # No response, declare as failed
+          if (!defined($reply) || !$reply) {
+               failed("updating %s: Could not connect to %s.", $h, $config{$h}{'server'});
+               last;
+          }
+          #last if !header_ok($h, $reply);
+
+          # Response found
+          if ($reply =~ /Success/i) {
+               $config{$h}{'ip'}     = $ip;
+               $config{$h}{'mtime'}  = $now;
+               $config{$h}{'status'} = 'good';
+               success("updating %s: %s: IP address set to %s", $h, 'good', $ip);
+         } else {
+               my ($statuscode, $statusmessage);
+               if($reply =~ /responseCode = (\d+)\nerrors_0_message = '(.*)'$/m) {
+                   $statuscode = $1;
+                   $statusmessage = $2;
+               }
+
+               $config{$h}{'status'} = 'failed';
+               failed("updating %s: %s - %s", $h, $statuscode, $statusmessage);
+          }
+     }
+}
+######################################################################
+## nic_dinahosting_examples
+######################################################################
+sub nic_dinahosting_examples {
```

Comment:
Move this function above `nic_dinahosting_update` (to match what other protocols do).